### DISCUSSION TITLE
Permissionless API for Filecoin

### DISCUSSION THREAD

**Discussion by poredamatt** [filecoin-project/community#560] [Developers]
Hi, is there a permissionless API allowing to pin files to IPFS and to pay Filecoin network for ensuring these files are stored even if nobody uses them? I.e. can I build on top of Filecoin (leveragin its storage incentives system on IPFS) without interacting with Filecoin's Business Development team? Can you point towards the right resources? Thank you!

**Comment by jennijuju** [comment]
Estuary.tech

**Reply by poredamatt** [reply]
Thanks, but Estuary is not permissionless - I need to request invite to start using it. Plus it says they only do public data storage for now. Is there any other API that actually allows to use Filecoin network without gatekeepers?

**Reply by jennijuju** [reply]
you can run your own shuttle / esturary nodes which don't require an invite, and you can store any data you want.

**Comment by jimmylee** [comment]
Estuary is permissionless, however

- https://estuary.tech is our example mainline service, this requires you to have an invite to use the account, we're happy to give you an invite key if you have a real use case. We do this to try to ensure a high quality service for everyone currently using us. That includes: ChainSafe, Zora, Mirror, etc.
- Running your own Estuary node starts here: https://github.com/application-research/estuary
  - You can store any data you want, to echo what @jennijuju said earlier
  - You can make modifications to your estuary node and do whatever you like
- You can have someone else run an Estuary node for you and they can lift the invite key requirement and make it available to everyone.

**Comment by turinglabsorg** [comment]
We are building a bunch of onchain and permissionless protocols. At the moment you can store on Filecoin through our Web3 Bounty Contract (live on Polygon) from here: https://web3bounty.app/, let me know if you need help!

Code is open-source and can be found here: https://github.com/protocol/onchain-storage/tree/main/protocols/web3storage-bounty

At the moment the only permissioned part is the one relative to the retrieval of filecoin's deals but we're working on it.

cc @irenegia @nicola @lucaniz
